src/vm/database/clarity_db.rs

```diff
@@ -263,20 +285,16 @@ impl <'a> ClarityDatabase <'a> {
 
         let data = DataMapMetadata { key_type, value_type };
 
-        let key = ClarityDatabase::make_key_for_trip(contract_identifier, StoreType::DataMapMeta, map_name);
-
-        assert!(!self.store.has_entry(&key), "Clarity VM attempted to initialize existing data map");
-
-        self.store.put(&key, &data.serialize());
+        let key = ClarityDatabase::make_metadata_key(StoreType::DataMapMeta, map_name);
```

Comment:
Cool, so we're making it so that the state root does not commit to data map or variable type metadata either?

Reply:
Correct -- no typing metadata is committed to.